File: solutions/go/resistor-color-duo/1/resistor_color_duo.go
package resistorcolorduo

var resistorColors = map[string]int{
    "black":  0,
    "brown":  1,
    "red":    2,
    "orange": 3,
    "yellow": 4,
    "green":  5,
    "blue":   6,
    "violet": 7,
    "grey":   8,
    "white":  9,
}

func colorNum(color string) int {
    return resistorColors[color]
}

// Value should return the resistance value of a resistor with a given colors.
func Value(colors []string) int {
	ohmQty := 0
    ohmQty += colorNum(colors[1]) + colorNum(colors[0]) * 10
    return ohmQty
}

File: solutions/csharp/resistor-color-duo/1/ResistorColorDuo.cs
public static class ResistorColorDuo
{
    private static readonly string[] colorCodes ={
         "black", "brown", "red", "orange", "yellow",
        "green", "blue", "violet", "grey", "white"
    };
    
    public static int Value(string[] colors)
    {
        int tens = ColorCode(colors[0]) * 10;
        int ones = ColorCode(colors[1]);
        return tens + ones;
    }

    public static int ColorCode(string color)
    {
        for (int i = 0; i < colorCodes.Length; i++)
        {
            if (colorCodes[i] == color)
            {
                return i;
            }
        }
        throw new ArgumentException("Invalid color");
    }
}
